src/sysusers/sysusers.c

```diff
@@ -1478,8 +1486,19 @@ static bool item_equal(Item *a, Item *b) {
         if (!streq_ptr(a->home, b->home))
                 return false;
 
-        if (!streq_ptr(a->shell, b->shell))
-                return false;
+        const char *a_shell = pick_shell(a),
+                   *b_shell = pick_shell(b);
+        if (!path_equal_ptr(a_shell, b_shell)) {
+                /* Let's check if the two paths refer to the same file.
+                 * For example, /sbin/nologin and /usr/sbin/nologin.
+                 * If we can't resolve something, treat different paths as different. */
+
+                const char *pa = prefix_roota(arg_root, a_shell),
+                           *pb = prefix_roota(arg_root, b_shell);
```

Comment:
for new code we really should use `chase_symlinks()` everywhere.

Specifically, use chase_symlinks_and_stat() here on the two paths, and then stat_inode_same().

Reply:
that said, I wonder if it wouldn't be better to ignore the status on disk. If this is really just about nologin, then maybe add an explicit check here somewhere:

```c
if (!path_equal_ptr(a_shell, b_shell) &&
    !(is_nologin_shell(a_shell) && is_nologin_shell(b_shell)) 
        return false;
```

benefit: /bin/true and /bin/false would be identified correctly as being equivalent, too.